include/openthread/srp_client.h

```diff
@@ -92,6 +92,7 @@ typedef struct otSrpClientService
 {
     const char *         mName;          ///< The service name labels (e.g., "_chip._udp", not the full domain name).
     const char *         mInstanceName;  ///< The service instance name label (not the full name).
+    const char *const *  mSubTypeLabels; ///< Array of service sub-type labels (must end with `NULL` or can be `NULL`).
```

Comment:
I am thinking if it is more convenient to encode the subtypes in `mName` or `mSubTypeLabels` by separating subtypes with `,`?

This reduces the number of buffers the user needs to pre-allocate and maintain.

Reply:
I considered this and I agree it may be easier for allocation on caller if all strings are dynamic. But I went for this model to keep `mName` and `mSubTypeLabeles` as separate strings and allow caller to use `const` strings for them.

I think service names (`mName`) are more often fixed/constant, so allowing them to use `const` string would be helpful, but the subtype labels may or may not be (e.g., I think in `matter`/CHIP most of the subtypes include some dynamic value). Thoughts?

Reply:
SGTM!